### PR TITLE
Style forms, add validation messages, sort gems

### PR DESCRIPTION
**Pull request by slpatrycja**

### CODE REVIEW

File: app/assets/stylesheets/lists.scss [hunk 1]
```diff
@@ -92,7 +92,42 @@ body, html {
...
+
+.tasks-container{
+  padding: 0 18px;
+  background-color: white;
+  max-height: 0;
+  overflow: hidden;
+  transition: max-height 0.2s ease-out;
+}
+
+.body {
```
Comment (Dariasz) [inline app/assets/stylesheets/lists.scss]: Body, do you mean html `body` or body class or body ? Please be more accurate with names

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -1,10 +1,9 @@
+<%= render "devise/shared/error_messages", resource: resource %>
```
Comment (Dariasz) [inline app/views/devise/registrations/new.html.erb]: Why not in slim ? Please don't support two template extensions

File: app/views/lists/edit.html.slim [hunk 3]
```diff
@@ -1,5 +1,9 @@
-h1
-  | Editing List
-= render 'form', list: @list
-= link_to 'Show', @list
-= link_to 'Back', lists_path
+- if @list.errors.any?
+  - @list.errors.full_messages.each do |message|
+    .error_card= message
+.body
+  .form
```
Comment (Dariasz) [inline app/views/lists/edit.html.slim]: is this `form` class needed?

File: app/views/lists/new.html.slim [hunk 4]
```diff
@@ -1,4 +1,9 @@
-h1
-  | New List
-= render 'form', list: @list
-= link_to 'Back', lists_path
+- if @list.errors.any?
+  - @list.errors.full_messages.each do |message|
+    .error_card= message
```
Comment (Dariasz) [inline app/views/lists/new.html.slim]: Please one convention the best BEM one

File: app/views/lists/new.html.slim [hunk 5]
```diff
@@ -1,4 +1,9 @@
...
+.body
+  .card
+    h1.header
+      | New List
```
Comment (Dariasz) [inline app/views/lists/new.html.slim]: we don't have to break a line

File: app/assets/stylesheets/lists.scss [hunk 6]
```diff
@@ -127,4 +162,34 @@ body, html {
 }
 
 
+.card {
```
Comment (Dariasz) [inline app/assets/stylesheets/lists.scss]: It is `card` class in `lists.scss` but it is also used in tasks view. Please rename or move to seperated file maybe in `components` directory.

Comment (Dariasz) [reply app/assets/stylesheets/lists.scss]: These styles are just duplicated please move it to component files

File: app/views/lists/edit.html.slim [hunk 7]
```diff
@@ -1,5 +1,9 @@
-h1
-  | Editing List
-= render 'form', list: @list
-= link_to 'Show', @list
-= link_to 'Back', lists_path
+- if @list.errors.any?
+  - @list.errors.full_messages.each do |message|
+    .error_card= message
```
Comment (Dariasz) [inline app/views/lists/edit.html.slim]: please one convention

File: app/views/tasks/_tasks_form.html.slim [hunk 8]
```diff
@@ -1,10 +1,2 @@
...
-      .task-container-item
-        - if task.in_progress?
-          = link_to fa_icon('exclamation')
-        -elsif task.done?
-          = link_to fa_icon('check')
-      .task-container-item
-        =task.name
\ No newline at end of file
+- list.tasks.each do |tasks|
+  =tasks.name
```
Comment (Dariasz) [inline app/views/tasks/_tasks_form.html.slim]: space after `=`

File: app/views/tasks/_form.html.slim [hunk 9]
```diff
@@ -10,9 +10,13 @@
             = message
   .field
```
Comment (Dariasz) [inline app/views/tasks/_form.html.slim]: These `br` 🤣 can we use `flex` and display it as a row ?

File: config/routes.rb [hunk 10]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  root to: 'lists#index'
```
Comment (Dariasz) [inline config/routes.rb]: thank you :)

File: app/views/lists/edit.html.slim [hunk 11]
```diff
@@ -1,5 +1,9 @@
-h1
-  | Editing List
-= render 'form', list: @list
-= link_to 'Show', @list
-= link_to 'Back', lists_path
+- if @list.errors.any?
```
Comment (Dariasz) [inline app/views/lists/edit.html.slim]: It is duplicated as well, why not partial ?

File: app/controllers/lists_controller.rb [hunk 12]
```diff
@@ -15,21 +15,19 @@ def edit; end
 
   def create
     @list = List.new(list_params)
-    if @list.save
-      redirect_to lists_path, notice: 'Task was successfully created.'
-    end
+    return redirect_to lists_path if @list.save
```
Comment (Dariasz) [inline app/controllers/lists_controller.rb]: do we need global variable here ?

File: app/assets/stylesheets/tasks.scss [hunk 13]
```diff
@@ -1,3 +1,39 @@
 // Place all the styles related to the tasks controller here.
```
Comment (Dariasz) [inline app/assets/stylesheets/tasks.scss]: I believe that we can remove these comments

File: app/views/devise/registrations/new.html.erb (hunk 2 above)
Comment (slpatrycja) [reply app/views/devise/registrations/new.html.erb]: Came in erb from devise gem, therefore stayed in erb :)

**Pull request merged**